e2e-tests/src/cardano_cli.py

```diff
@@ -88,19 +124,24 @@ def generate_stake_keys(self):
             logger.error(result.stderr)
             return None, None
 
-        # Convert to a valid JSON array
-        modified_response = "[" + result.stdout.replace("\r", "").replace("}\n{", "},\n{") + "]"
-        parsed_data = json.loads(modified_response)
-
-        signing_key = parsed_data[0]
-        verification_key = parsed_data[1]
-
-        logger.info(f"Stake signing key: {signing_key}")
-        logger.info(f"Stake verification key: {verification_key}")
+        signing_key, verification_key = self.__parse_keys_from_stdout(result.stdout)
+        logger.debug(f"Stake signing key: {signing_key}")
+        logger.debug(f"Stake verification key: {verification_key}")
         return signing_key, verification_key
 
+    def build_address(self, payment_vkey, stake_vkey=None):
```

Comment:
Key file contains the information if it is:
*stake pool operator (however IDK if we can make address for it)
*staking
* payment
, so the parameter could be removed. More implementation here gives API that is harder to feed with invalid parameters.